src/nbl/video/IOpenGL_PhysicalDeviceBase.h

```diff
@@ -561,11 +561,52 @@ class IOpenGL_PhysicalDeviceBase : public IOpenGLPhysicalDeviceBase
 		}
 		else m_glfeatures.MaxAnisotropy = 0u;
 
+		m_features.fullDrawIndexUint32 = false; // TODO
+		m_features.independentBlend = false; // TODO
 		if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_ARB_geometry_shader4))
 		{
-			GetIntegerv(GL_MAX_GEOMETRY_OUTPUT_VERTICES, &num);
-			m_glfeatures.MaxGeometryVerticesOut = static_cast<uint32_t>(num);
 			m_features.geometryShader = true;
+			GetIntegerv(GL_MAX_GEOMETRY_SHADER_INVOCATIONS, reinterpret_cast<GLint*>(&m_properties.limits.maxGeometryShaderInvocations));
+			GetIntegerv(GL_MAX_GEOMETRY_INPUT_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxGeometryInputComponents));
+			GetIntegerv(GL_MAX_GEOMETRY_OUTPUT_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxGeometryOutputComponents));
+			GetIntegerv(GL_MAX_GEOMETRY_OUTPUT_VERTICES, reinterpret_cast<GLint*>(&m_properties.limits.maxGeometryOutputVertices));
+			GetIntegerv(GL_MAX_GEOMETRY_TOTAL_OUTPUT_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxGeometryTotalOutputComponents));
+			m_glfeatures.MaxGeometryVerticesOut = m_properties.limits.maxGeometryOutputVertices;
+		}
+		if (m_glfeatures.isFeatureAvailable(m_glfeatures.NBL_ARB_tessellation_shader))
+		{
+			m_features.tessellationShader = true;
+			GetIntegerv(GL_MAX_TESS_GEN_LEVEL, reinterpret_cast<GLint*>(&m_properties.limits.maxTessellationGenerationLevel));
+			// GL_MAX_PATCH_VERTICES => maximum patch size
+			GetIntegerv(GL_MAX_PATCH_VERTICES, reinterpret_cast<GLint*>(&m_properties.limits.maxTessellationPatchSize));
+			// GL_MAX_TESS_CONTROL_INPUT_COMPONENTS => num. components for per-vertex inputs in tess.
+			GetIntegerv(GL_MAX_TESS_CONTROL_INPUT_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxTessellationControlPerVertexInputComponents));
+			// GL_MAX_TESS_CONTROL_OUTPUT_COMPONENTS => num. components for per-vertex outputs in tess. 
+			GetIntegerv(GL_MAX_TESS_CONTROL_OUTPUT_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxTessellationControlPerVertexOutputComponents)); 
+			// GL_MAX_TESS_PATCH_COMPONENTS => num. components for per-patch output varyings
+			GetIntegerv(GL_MAX_TESS_PATCH_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxTessellationControlPerPatchOutputComponents));
+			GetIntegerv(GL_MAX_TESS_CONTROL_TOTAL_OUTPUT_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxTessellationControlTotalOutputComponents));
+			GetIntegerv(GL_MAX_TESS_EVALUATION_INPUT_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxTessellationEvaluationInputComponents));
+			GetIntegerv(GL_MAX_TESS_EVALUATION_OUTPUT_COMPONENTS, reinterpret_cast<GLint*>(&m_properties.limits.maxTessellationEvaluationOutputComponents));
+		}
+
+		m_features.drawIndirectFirstInstance = false; // TODO
```

Comment:
I think both GL and GLES (but only whenever GLES supports `drawIndirect`) support this

Reply:
`IsGLES ? (m_glfeatures.isFeatureAvailable(COpenGLFeatureMap::NBL_EXT_multi_draw_indirect)):true`